### PR TITLE
audit: Don't run audit methods when `--only-cops` option is passed

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Previously `--only-cops` in addition to the passed cops, would also run audit methods defined in `FormulaAuditor`. This PR would change that to run only the cops passed through `--only-cops` and `audit_style` audit method to report style offenses.

Also while testing I discovered that I forgot to commit `.rubocop.yml` and `rubocops.rb` in #2790 
So adding those files here too so `LegacyPatches` cop would be enabled.

### CODE REVIEW

File: Library/Homebrew/cmd/style.rb [hunk 1]
```diff
@@ -94,7 +94,12 @@ def check_style_impl(files, output_type, options = {})
           RuboCop::Cop::Cop.registry.departments.include?(cop.to_sym)
       end
 
-      args << "--only" << cops_to_include.join(",") unless cops_to_include.empty?
+      if cops_to_include.empty?
+        odie "Cop(s): #{options[:only_cops].join(",")} were either not found/enabled"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: `RuboCops #{options[:only_cops].join(",")} were not found`

File: Library/Homebrew/cmd/style.rb [hunk 2]
```diff
@@ -94,7 +94,12 @@ def check_style_impl(files, output_type, options = {})
...
+      else
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: don't need the `else` as `odie` will leave the function.

**Comment by GauthamGoli** [comment]
`brew audit --only-cops=Desc` 
`brew audit --only-cops=ComponentsOrder`
Commands like these used to run other non cop checks too so the command used to take a lot of time. Now the command just runs the cop and no other check.

**Comment by MikeMcQuaid** [comment]
Thanks again @GauthamGoli!

**Comment by JCount** [comment]
@GauthamGoli 👍

**Pull request merged**